Packer crash when cancelling Proxmox builder

Packer crashes almost every time when when cancelling Proxmox builder.

- Debug log output from `PACKER_LOG=1 packer build template.json`.

- 3 (out of 9) crash logs and the template:
https://gist.github.com/chriswayg/c24b4577d99842ebe658d1576736c825

- a relevant part:
> 2019/08/27 12:07:50 packer: 2019/08/27 12:07:50 [DEBUG] Error getting SSH address: 500 QEMU guest agent is not running
> 2019/08/27 12:07:57 ui error: Cancelling build after receiving interrupt
> 2019/08/27 12:07:57 packer: 2019/08/27 12:07:57 Received interrupt signal (count: 1). Ignoring.
> 2019/08/27 12:07:57 Cancelling builder after context cancellation context canceled
> 2019/08/27 12:07:57 packer: 2019/08/27 12:07:57 Received interrupt signal (count: 1). Ignoring.
> 2019/08/27 12:07:57 packer: 2019/08/27 12:07:57 [WARN] Interrupt detected, quitting waiting for SSH.
> 2019/08/27 12:07:57 ui: [1;32m==> proxmox: Stopping VM[0m
> 2019/08/27 12:07:57 packer: 2019/08/27 12:07:57 [DEBUG] Error getting SSH address: 500 QEMU guest agent is not running
> 2019/08/27 12:07:57 packer: 2019/08/27 12:07:57 [DEBUG] SSH wait cancelled. Exiting loop.
> 2019/08/27 12:07:59 ui: [1;32m==> proxmox: Deleting VM[0m
> 2019/08/27 12:08:01 packer: panic: interface conversion: interface {} is nil, not int
> 2019/08/27 12:08:01 packer: 
> 2019/08/27 12:08:01 packer: goroutine 160 [running]:
> 2019/08/27 12:08:01 packer: github.com/hashicorp/packer/builder/proxmox.(*Builder).Run(0xc000385b00, 0x3fadba0, 0xc0005da040, 0x3fc97a0, 0xc0005d6040, 0x3f46f40, 0xc0005fe010, 0x40b9b9, 0xc0005f8b70, 0x30, ...)
> 2019/08/27 12:08:01 packer: 	/Users/mmarsh/Projects/packer/builder/proxmox/builder.go:94 +0xdb9
> 2019/08/27 12:08:01 packer: github.com/hashicorp/packer/packer/rpc.(*BuilderServer).Run(0xc0004722d0, 0x1, 0xc00058d2d0, 0x0, 0x0)
> 2019/08/27 12:08:01 packer: 	/Users/mmarsh/Projects/packer/packer/rpc/builder.go:109 +0x231
> 2019/08/27 12:08:01 packer: reflect.Value.call(0xc000594360, 0xc0003a6220, 0x13, 0x3998dd5, 0x4, 0xc0000d8f18, 0x3, 0x3, 0xc00005c200, 0x0, ...)
> 2019/08/27 12:08:01 packer: 	/usr/local/go/src/reflect/value.go:447 +0x461
> 2019/08/27 12:08:01 packer: reflect.Value.Call(0xc000594360, 0xc0003a6220, 0x13, 0xc00051ff18, 0x3, 0x3, 0x0, 0x0, 0x0)
> 2019/08/27 12:08:01 packer: 	/usr/local/go/src/reflect/value.go:308 +0xa4
> 2019/08/27 12:08:01 packer: net/rpc.(*service).call(0xc0003a40c0, 0xc0000e40a0, 0xc0000500c8, 0xc0000500d0, 0xc00016c100, 0xc00000c1e0, 0x2ecd180, 0xc00058d2ac, 0x18a, 0x2e1fce0, ...)
> 2019/08/27 12:08:01 packer: 	/usr/local/go/src/net/rpc/server.go:384 +0x14e
> 2019/08/27 12:08:01 packer: created by net/rpc.(*Server).ServeCodec
> 2019/08/27 12:08:01 packer: 	/usr/local/go/src/net/rpc/server.go:481 +0x42b
> 2019/08/27 12:08:01 [INFO] (telemetry) ending proxmox
> 2019/08/27 12:08:01 ui error: [1;31mBuild 'proxmox' errored: unexpected EOF[0m
> 2019/08/27 12:08:01 ui: Cleanly cancelled builds after being interrupted.

- Packer version from `packer version`
- Host platform
```
packer version && pveversion && lsb_release -d && cat /etc/debian_version
		Packer v1.4.3
		pve-manager/6.0-5/f8a710d7 (running kernel: 5.0.18-1-pve)
		Description:	Debian GNU/Linux 10 (buster)
		10.0
```

Hi @azr, any fix yet? - Its really annoying to have Packer crash all the time ;-) You probably don't need any more crash reports... They all look like they crash at the same spot, (to me anyways).


```
proxmox output will be in this color.

==> proxmox: Creating VM
==> proxmox: Starting VM
==> proxmox: Starting HTTP server on port 8642
==> proxmox: Waiting 10s for boot
==> proxmox: Typing the boot command
==> proxmox: Waiting for SSH to become available...
Cancelling build after receiving interrupt
==> proxmox: Stopping VM
==> proxmox: Deleting VM
Build 'proxmox' errored: unexpected EOF
Cleanly cancelled builds after being interrupted.
panic: interface conversion: interface {} is nil, not int
2019/09/04 17:48:14 packer: 
2019/09/04 17:48:14 packer: goroutine 182 [running]:
2019/09/04 17:48:14 packer: github.com/hashicorp/packer/builder/proxmox.(*Builder).Run(0xc00047e480, 0x3fadba0, 0xc0005e8080, 0x3fc97a0, 0xc00039c120, 0x3f46f40, 0xc0003783a0, 0x40b9b9, 0xc0005efe30, 0x30, ...)
2019/09/04 17:48:14 packer: 	/Users/mmarsh/Projects/packer/builder/proxmox/builder.go:94 +0xdb9
2019/09/04 17:48:14 packer: github.com/hashicorp/packer/packer/rpc.(*BuilderServer).Run(0xc00049ccf0, 0x1, 0xc00039ac60, 0x0, 0x0)
2019/09/04 17:48:14 packer: 	/Users/mmarsh/Projects/packer/packer/rpc/builder.go:109 +0x231
2019/09/04 17:48:14 packer: reflect.Value.call(0xc0002e0960, 0xc00039e098, 0x13, 0x3998dd5, 0x4, 0xc0000d5f18, 0x3, 0x3, 0xc000062580, 0x0, ...)
2019/09/04 17:48:14 packer: 	/usr/local/go/src/reflect/value.go:447 +0x461
2019/09/04 17:48:14 packer: reflect.Value.Call(0xc0002e0960, 0xc00039e098, 0x13, 0xc000508718, 0x3, 0x3, 0x0, 0x0, 0x0)
2019/09/04 17:48:14 packer: 	/usr/local/go/src/reflect/value.go:308 +0xa4
2019/09/04 17:48:14 packer: net/rpc.(*service).call(0xc00038e300, 0xc0003aa280, 0xc00046de18, 0xc00046de20, 0xc0003a4500, 0xc0002d9360, 0x2ecd180, 0xc00039ac3c, 0x18a, 0x2e1fce0, ...)
2019/09/04 17:48:14 packer: 	/usr/local/go/src/net/rpc/server.go:384 +0x14e
2019/09/04 17:48:14 packer: created by net/rpc.(*Server).ServeCodec
2019/09/04 17:48:14 packer: 	/usr/local/go/src/net/rpc/server.go:481 +0x42b
2019/09/04 17:48:14 [INFO] (telemetry) ending proxmox
2019/09/04 17:48:14 /usr/local/bin/packer: plugin process exited
2019/09/04 17:48:14 ui error: Build 'proxmox' errored: unexpected EOF
2019/09/04 17:48:14 ui: Cleanly cancelled builds after being interrupted.
2019/09/04 17:48:14 [INFO] (telemetry) Finalizing.
2019/09/04 17:48:14 waiting for all plugin processes to complete...
2019/09/04 17:48:14 /usr/local/bin/packer: plugin process exited
2019/09/04 17:48:14 /usr/local/bin/packer: plugin process exited



!!!!!!!!!!!!!!!!!!!!!!!!!!! PACKER CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

Packer crashed! This is always indicative of a bug within Packer.
A crash log has been placed at "crash.log" relative to your current
working directory. It would be immensely helpful if you could please
report the crash with Packer[1] so that we can fix this.

[1]: https://github.com/hashicorp/packer/issues

!!!!!!!!!!!!!!!!!!!!!!!!!!! PACKER CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!
```

--- a/builder/proxmox/builder.go
+++ b/builder/proxmox/builder.go
@@ -3,6 +3,7 @@ package proxmox
 import (
 	"context"
 	"crypto/tls"
+	"errors"
 	"fmt"
 
 	"github.com/Telmate/proxmox-api-go/proxmox"
@@ -89,7 +90,12 @@ func (b *Builder) Run(ctx context.Context, ui packer.Ui, hook packer.Hook) (pack
 	if rawErr, ok := state.GetOk("error"); ok {
 		return nil, rawErr.(error)
 	}
+	// If we were interrupted or cancelled, then just exit.
+	if _, ok := state.GetOk(multistep.StateCancelled); ok {
+		return nil, errors.New("build was cancelled")
+	}
 
+	// Verify that the template_id was set properly, otherwise we didn't progress through the last step
 	tplID, ok := state.Get("template_id").(int)
 	if !ok {
 		return nil, fmt.Errorf("template ID could not be determined")

